fastlane/actions/get_latest_github_release_within_same_major_action.rb

```diff
@@ -0,0 +1,99 @@
+module Fastlane
+  module Actions
+
+    class GetLatestGithubReleaseWithinSameMajorAction < Action
+      def self.run(params)
+        repo_name = params[:repo_name]
+        UI.message("Getting latest release for #{repo_name} on GitHub within same major as the current version")
+
+        GithubApiAction.run(
+          server_url: "https://api.github.com",
+          api_token: ENV["GITHUB_API_TOKEN"],
```

Comment:
This could be set as an optional parameter, similar to how I do it here: https://github.com/RevenueCat/fastlane-plugin-revenuecat_internal/blob/main/lib/fastlane/plugin/revenuecat_internal/actions/bump_version_update_changelog_create_pr_action.rb#L89-L93

Also, I think it's not really needed since our repos are public so not sure if we want to remove it entirely (we should test if we decide to remove it)

Reply:
Note this is me calling another action, not part of the action's parameters. In any case, I thought it was required by `github_action` but it doesn't look like it is? https://docs.fastlane.tools/actions/github_api/ 

Maybe we can just remove it?

Reply:
Ok, I see what you mean. What would be the advantage of setting it as an optional parameter versus removing it? What should we test if we decide to remove it?

Reply:
Well setting it as an optional parameter would allow callers to pass the token either as a parameter or through an environment variable. This would be needed if we wanted to access private repos if we ever need to in the future. Besides that, authenticated requests have some advantages such as increased rate limits. 

That said, i don't think those factors should affect us... So i might just remove it to keep it simple. As long as it works without any token.

Reply:
I removed it along with the bearer. We can always bring them back if we need to